src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -138,34 +148,32 @@ CHIP_ERROR Type::Encode(TLV::TLVWriter & aWriter, TLV::Tag aTag) const{
     ReturnErrorOnFailure(DataModel::Encode(aWriter, TLV::ContextTag(Fields::k{{asUpperCamelCase name}}), {{asLowerCamelCase name}}));
     {{/if_is_fabric_scoped_struct}}
     {{/zcl_event_fields}}
-    ReturnErrorOnFailure(aWriter.EndContainer(outer));
-    return CHIP_NO_ERROR;
+    return aWriter.EndContainer(outer);
 }
 
 CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
-    TLV::TLVType outer;
-    ReturnErrorOnFailure(chip::app::Clusters::detail::FlightCheckDecodeAndEnterStruct(reader, outer));
+    detail::StructDecodeIterator __iterator(reader);
+    while (true) {
+        auto __element = __iterator.Next();
+        if (std::holds_alternative<CHIP_ERROR>(__element)) {
+           return std::get<CHIP_ERROR>(__element);
+        }
+
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        [[maybe_unused]] const uint8_t __context_tag = std::get<uint8_t>(__element);
```

Comment:
This and structs don't seem to be using the #first thing... just commands are?

Reply:
oops, missed these ones.
Structs always have members so they do not benefit much, however events maybe benefit and everything likes consistency.